ext/lmdb-safe/lmdb-safe.hh

```diff
@@ -157,15 +267,19 @@ public:
   {
   }
 
+#ifndef DNSDIST
   template <class T,
-            typename std::enable_if<std::is_arithmetic<T>::value,
+            typename std::enable_if<std::is_integral<T>::value,
                                     T>::type* = nullptr>
   MDBInVal(T i)
   {
-    memcpy(&d_memory[0], &i, sizeof(i));
+    auto j = htonl(i);    // all actual usage in our codebase is 32 bits. If that ever changes, this will break the build and avoid runtime surprises
```

Comment:
I would really prefer a `static_assert(sizeof(T) == 4, "...")` if we want to rely on that.

Reply:
I'm really wondering if this is the right place to do that conversion, though, because it makes an assumption about the data that is read that might be application-dependent.

Reply:
added the assert.

I agree, it's not the best place, but in the context of our code base, it is currently the simplest place.. (that said, if this really was uint32_t, most applications -would- prefer storage be in network order - and of the two current consumers of this copy of lmdb-safe, one (auth) has conversion in this version, and the other (dnsdist) currently does not read numbers this way)